packages/root/src/Root.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+
+type ButtonProps<T> = HTMLElementProps<'button'> &
+  T & {
+    href?: undefined;
+  };
+
+type Anchor<T> = HTMLElementProps<'a'> &
+  T & {
+    href: string;
+  };
+
+type CustomElementProps<T> = T & {
+  as: React.ElementType<any>;
+  [key: string]: any;
+};
+
+type RootProps<T> = ButtonProps<T> | Anchor<T> | CustomElementProps<T>;
+
+function usesCustomElement<T>(
+  props: RootProps<T>,
+): props is CustomElementProps<T> {
+  return (props as any).as != null;
+}
+
+function usesLinkElement<T>(props: RootProps<T>): props is Anchor<T> {
+  return props.href != null;
+}
+
+/**
+ * # Root
+ *
+ * Root component
+ *
+ * ```
+<Root href="https://mongodb.design">Anchors Away!</Root>
+```
+ * @param props.children Takes priority in determining what tag the `<Root />` element will render.
+ * @param props.href Determines the location that an anchor tag will point to. If prop is set without the `as` prop, `<Root />` will render `a` tags.
+ * @param props.as Callback to change the open state of the Menu.
+ */
+function Root<T>(props: RootProps<T>) {
```

Comment:
I feel conflicted about calling this "Root". My first instinct would be for a `Root` component to go at the top level of your app or something. Since this defaults to rendering a `button`, maybe `BaseButton` would be a better name? Not 100% confident on that, but `Root` seems to imply a bunch of things that aren't true to me.

Reply:
i can understand that Root might be confusing. Thoughts @DesignerDave and @seanmhanson ?

Reply:
I think that even though it defaults to button, we shouldn't name it after button (half the usages of this would have a menuitem role, and half would have a button role, for example). I've referred to it as Root so often that I can't un-think that name right now, but I agree that Root falls into that category with Application, App, etc.

Reply:
I really like both the suggestion of `Box` and of `component` instead of `as`. I also prefer `div` as a default, but Brooke made a good case for `button` since this is more of an internal component.

Reply:
Should we call it Component? Reach UI has a Component component and maybe that's more clear? We can default to div if that makes people feel more comfortable we've just followed the Root/button pattern it all of the use-cases thus far

Reply:
I feel like when we make a box component it should have more grid/spacing/alignment properties built in @hswolff, but maybe I'm wrong here

Reply:
Naming this Box now doesn't preclude us from adding those props at a future point. The Box component would also have this functionality of the `component={}` prop so this gets us started down that path.

Might require a more rapid discussion on Slack.

Reply:
Renamed to Box!

Reply:
(with div as default, and component as slot instead of as)